homeassistant/components/binary_sensor/esphome.py

```diff
@@ -0,0 +1,63 @@
+"""Support for ESPHome binary sensors."""
+import logging
+
+from typing import TYPE_CHECKING, Optional
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.esphome import EsphomeEntity, \
+    platform_async_setup_entry
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from aioesphomeapi import BinarySensorInfo, BinarySensorState  # noqa
+
+DEPENDENCIES = ['esphome']
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up ESPHome binary sensors based on a config entry."""
+    # pylint: disable=redefined-outer-name
+    from aioesphomeapi import BinarySensorInfo, BinarySensorState  # noqa
+
+    await platform_async_setup_entry(
+        hass, entry, async_add_entities,
+        component_key='binary_sensor',
+        info_type=BinarySensorInfo, entity_type=EsphomeBinarySensor,
+        state_type=BinarySensorState
+    )
+
+
+class EsphomeBinarySensor(EsphomeEntity, BinarySensorDevice):
+    """A binary sensor implementation for ESPHome."""
+
+    @property
+    def _static_info(self) -> 'BinarySensorInfo':
+        return super()._static_info
```

Comment:
Why implement this if you're super'ing it?

Reply:
See https://github.com/home-assistant/home-assistant/pull/19334#discussion_r241996357 - it's for static typing.

I tried to do it like this too:

```python
@property
def _static_info(self) -> 'BinarySensorInfo': ...
```

with variations of including/excluding `@property` and `@overload` but I couldn't get it to work.